### PR TITLE
OE  view footers POM

### PR DESCRIPTION
**Pull request by OxanaE**
Refactored test viewFooter into POM. 
https://trello.com/c/UhoQMqiG/674-entity-footers-pom-viewfooter
But the tusk is incomplete at the moment because of the bug in the app.
https://trello.com/c/TWFyzEdQ/106-entityfooters-control-values-are-not-displayed-on-view-page
Also rearranged tests in the class to use RunType.Multiple.

Ignored installFromTemplateTest, portal.MarketplaceTemplateTest
https://trello.com/c/gO6f3Yee/821-installfromtemplatetest-portalmarketplacetemplatetest

### CODE REVIEW

File: src/test/java/test/entity/EntityFootersTest.java [hunk 1]
```diff
@@ -78,41 +70,28 @@
...
-            Assert.assertEquals(columns.size(), values.length);
-
-        for (int i = 0; i < columns.size(); i++) {
-            if (values[i] != null) {
-                Assert.assertEquals(columns.get(i).getText(), values[i]);
-            }
-        }
+    @Test
+    public void sumFooter() {
+        FootersEditPage footersEditPage = new MainPage(getDriver())
```
Comment (MaxBurdin) [inline src/test/java/test/entity/EntityFootersTest.java:78]: You don't use this variable anywhere. Why do you need it?

**Review by MaxBurdin — APPROVED** [review]

File: src/test/java/test/entity/EntityFootersTest.java [hunk 2]
```diff
@@ -78,41 +70,28 @@
...
-        for (int i = 0; i < columns.size(); i++)
-            Assert.assertEquals(columns.size(), values.length);
-
-        for (int i = 0; i < columns.size(); i++) {
-            if (values[i] != null) {
-                Assert.assertEquals(columns.get(i).getText(), values[i]);
-            }
-        }
+    @Test
+    public void sumFooter() {
```
Comment (oooleks) [inline src/test/java/test/entity/EntityFootersTest.java:77]: "Test" word is missing in every test name;
There is no any Assert in test, so what it tests?

File: src/test/java/test/entity/EntityFootersTest.java [hunk 3]
```diff
@@ -78,41 +70,28 @@
...
+                .fillSumFootersRowData(2, INT_VALUE_3, DEC_VALUE_3)
+                .waitSumFootersControlToBe(SUM_RESULT)
+                .deleteRow(SUM_FOOTERS, 3)
+                .clickPlusSumButton()
+                .fillSumFootersRowData(3, INT_VALUE_4, DEC_VALUE_4)
+                .waitSumFootersControlToBe(SUM_RESULT_DELETED_ROW);
     }
 
-    @Test
+    @Test(dependsOnMethods = "sumFooter")
```
Comment (oooleks) [inline src/test/java/test/entity/EntityFootersTest.java:94]: There is no any Assert in test.
We do not place any methods into POM files to replace Assert.

**Review by guram21 — APPROVED** [review]

File: src/test/java/test/entity/EntityFootersTest.java [hunk 4]
```diff
@@ -135,7 +114,7 @@ public void minFooter() {
                 .waitMinFootersControlToBe(MIN_RESULT_DELETED_ROW);
     }
 
-    @Test
+    @Test(dependsOnMethods = "minFooter")
```
Comment (oooleks) [inline src/test/java/test/entity/EntityFootersTest.java:117]: There is no any Assert in test, so what it tests?

File: src/test/java/test/entity/EntityFootersTest.java [hunk 5]
```diff
@@ -279,7 +215,7 @@ public void deleteFooter() {
         Assert.assertEquals(footersPage.getRowCount(), 0);
     }
 
-    @Test
+    @Test(dependsOnMethods = "deleteFooter")
```
Comment (oooleks) [inline src/test/java/test/entity/EntityFootersTest.java:218]: There is no any Assert in test, so what it tests?

**Review by oooleks — CHANGES_REQUESTED** [review]

File: src/test/java/test/entity/EntityFootersTest.java (hunk 1 above)
Comment (OxanaE) [reply src/test/java/test/entity/EntityFootersTest.java:78]: Sergey was teaching us this way.

File: src/test/java/test/entity/EntityFootersTest.java (hunk 2 above)
Comment (OxanaE) [reply src/test/java/test/entity/EntityFootersTest.java:77]: I understand it's not the requirement, only class name should have the word Test in its name. It's working just fine without it.

File: src/test/java/test/entity/EntityFootersTest.java (hunk 3 above)
Comment (OxanaE) [reply src/test/java/test/entity/EntityFootersTest.java:94]: `public FootersEditPage waitMinFootersControlToBe(String numbers) {
        getWait().until(d -> minControl.getAttribute("value").contains(numbers));
        return this;
    }`
The system waits until a certain condition is met, this is the validation.

File: src/test/java/test/entity/EntityFootersTest.java (hunk 2 above)
Comment (oooleks) [reply src/test/java/test/entity/EntityFootersTest.java:77]: NP, yes, you are right - it's working just fine, but to be consistent with our framework and as a good practice we use it. Or for small like unit tests we can name them using the pattern [MethodUnderTest]_[Scenario]_[ExpectedResult] (e.g.: Sum_TwoNumbers_ReturnsSum()). 
Could you comment on assert item? Thanks.

**Comment by OxanaE** [comment]
I was just not aware of this rule about naming tests. Maybe we should have
some document as a team with all the coding rules. I know I for sure have
many questions about them.
I commented on the assert issue in PR as well (the first one out of three).
So the idea is because we are using "wait until a certain condition is met"
it takes a place of our usual validation. If a certain value (control
sequence) is not displayed in the field, the test will fail.

**Comment by guram21** [comment]
Good point about having a document with "the coding rules."  We will discuss it on today's stand-up.

**Review by oooleks — APPROVED** [review]

**Review by homelyn-code — CHANGES_REQUESTED** [review]
Please add assertion after saving the record. It should not be any validation during creation or editing.

**Pull request merged**